File: mobile-ios/Models/RAGModels.swift
import Foundation

struct RAGMatch: Identifiable, Codable {
    var id: String { documentId }
    let documentId: String
    let title: String
    let category: String
    let relevanceScore: Double
    let matchPercentage: Double
    let snippet: String

    enum CodingKeys: String, CodingKey {
        case documentId = "document_id"
        case title
        case category
        case relevanceScore = "relevance_score"
        case matchPercentage = "match_percentage"
        case snippet
    }
}

struct RAGResponse: Codable {
    let query: String
    let aiSynthesis: String
    let vectorDimension: Int
    let totalDocumentsIndexed: Int
    let topMatches: [RAGMatch]

    enum CodingKeys: String, CodingKey {
        case query
        case aiSynthesis = "ai_synthesis"
        case vectorDimension = "vector_dimension"
        case totalDocumentsIndexed = "total_documents_indexed"
        case topMatches = "top_matches"
    }
}

struct RAGAskRequest: Codable {
    let query: String
}

struct RAGUploadResponse: Codable {
    let filename: String
    let documentId: String
    let chunksCreated: Int
    let vectorDimension: Int
    let status: String
    let message: String

    enum CodingKeys: String, CodingKey {
        case filename
        case documentId = "document_id"
        case chunksCreated = "chunks_created"
        case vectorDimension = "vector_dimension"
        case status
        case message
    }
}

enum EventSeverity: String, Codable, CaseIterable {
    case info = "Info"
    case warning = "Warning"
    case critical = "Critical"

    var colorName: String {
        switch self {
        case .info: return "blue"
        case .warning: return "orange"
        case .critical: return "red"
        }
    }
}

struct TelemetryEvent: Identifiable, Codable {
    let id: String
    let timestamp: String
    let sourceComponent: String
    let category: String
    let severity: EventSeverity
    let message: String
    let executionTimeMs: Double?

    enum CodingKeys: String, CodingKey {
        case id
        case timestamp
        case sourceComponent = "source_component"
        case category
        case severity
        case message
        case executionTimeMs = "execution_time_ms"
    }
}

File: mobile-android/app/src/main/java/com/nexusops/model/RAGModels.kt
package com.nexusops.model

data class RAGMatchItem(
    val document_id: String,
    val title: String,
    val category: String,
    val relevance_score: Double,
    val match_percentage: Double,
    val snippet: String
)

data class RAGResponseItem(
    val query: String,
    val ai_synthesis: String,
    val vector_dimension: Int,
    val total_documents_indexed: Int,
    val top_matches: List<RAGMatchItem>
)

data class RAGAskPayload(
    val query: String
)

data class RAGUploadResponse(
    val filename: String,
    val document_id: String,
    val chunks_created: Int,
    val vector_dimension: Int,
    val status: String,
    val message: String
)

enum class EventSeverity(val label: String) {
    INFO("Info"),
    WARNING("Warning"),
    CRITICAL("Critical")
}

data class TelemetryEvent(
    val id: String,
    val timestamp: String,
    val sourceComponent: String,
    val category: String,
    val severity: EventSeverity,
    val message: String,
    val executionTimeMs: Double? = null
)

